src/web/components/StickyBottomBanner/ReaderRevenueBanner.tsx

```diff
@@ -157,96 +168,164 @@ export const canShow = async ({
 	return getBanner(
 		bannerPayload,
 		`${contributionsServiceUrl}/banner${queryString}`,
-	)
-		.then(checkForErrors)
-		.then((response) => response.json())
-		.then((json: { data?: any }) => {
+	).then((json: { data?: any }) => {
+		if (!json.data) {
+			if (
+				engagementBannerLastClosedAt &&
+				subscriptionBannerLastClosedAt
+			) {
+				setLocalNoBannerCachePeriod();
+			}
+			return { result: false };
+		}
+
+		const { module, meta } = json.data;
+
+		return { result: true, meta: { module, meta } };
+	});
+};
+
+export const canShowPuzzlesBanner: CanShowFunction = async ({
+	remoteBannerConfig,
+	isSignedIn,
+	asyncCountryCode,
+	contentType,
+	sectionName,
+	shouldHideReaderRevenue,
+	isMinuteArticle,
+	isPaidContent,
+	isSensitive,
+	tags,
+	contributionsServiceUrl,
+	alreadyVisitedCount,
+	engagementBannerLastClosedAt,
+	subscriptionBannerLastClosedAt,
+	section,
+}) => {
+	const isPuzzlesPage =
+		section === 'crosswords' ||
+		tags.some((tag) => tag.type === 'Series' && tag.title === 'Sudoku');
+
+	if (shouldHideReaderRevenue) {
+		// We never serve Reader Revenue banners in this case
+		return { result: false };
+	}
+
+	if (isPuzzlesPage && remoteBannerConfig) {
+		const countryCode = await asyncCountryCode;
+		const hasConsentedToArticleCounts = await getArticleCountConsent();
+		const bannerPayload = buildPayload({
+			isSignedIn,
+			countryCode,
+			contentType,
+			sectionName,
+			shouldHideReaderRevenue,
+			isMinuteArticle,
+			isPaidContent,
+			tags,
+			contributionsServiceUrl,
+			isSensitive,
+			alreadyVisitedCount,
+			engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt,
+			hasConsentedToArticleCounts,
+		});
+		return getBanner(
+			bannerPayload,
+			`${contributionsServiceUrl}/puzzles`,
+		).then((json: { data?: any }) => {
 			if (!json.data) {
-				if (
-					engagementBannerLastClosedAt &&
-					subscriptionBannerLastClosedAt
-				) {
-					setLocalNoBannerCachePeriod();
-				}
 				return { result: false };
 			}
 
 			const { module, meta } = json.data;
 
 			return { result: true, meta: { module, meta } };
 		});
+	}
+
+	return { result: false };
 };
 
-type Props = {
+export type BannerProps = {
 	meta: any;
 	module: { url: string; name: string; props: any[] };
 };
 
-export const ReaderRevenueBanner = ({ meta, module }: Props) => {
-	const [Banner, setBanner] = useState<React.FC>();
+const createReaderRevenueBannerComponent = (
```

Comment:
I read this component more as just a Base Banner component than a Reader Revenue Banner? Maybe it might be better to use a different name to indicate it is a more generic component?

Reply:
I don't want to make it _too_ generic in case that might create confusion with the Braze banners which work in a very different way, but this is the pattern we'd use for anything coming from the `support-dotcom-components` service. Maybe it's a `RemoteBanner`?

Reply:
Remote banner sounds like a good name, I am just keen to drop `ReaderRevenue` from the name